ui/apps/platform/src/Containers/Policies/Wizard/Step3/PolicyCriteriaOptions.tsx

```diff
@@ -0,0 +1,188 @@
+    function onSearch(evt) {
+        const input: string = evt.target.value;
+        if (input === '') {
+            setFilteredItems(treeDataItems);
+            setIsFiltered(false);
+        } else {
+            const filtered = treeDataItems.map((item) => {
+                const filteredItem = { ...item };
+                if (item.children && item.children.length > 0) {
+                    const filteredChildren = item.children.filter((child) => {
+                        const name = typeof child.name === 'string' ? child.name : '';
+                        const title = typeof child.title === 'string' ? child.title : '';
+                        return (
+                            name.toLowerCase().includes(input.toLowerCase()) ||
+                            title.toLowerCase().includes(input.toLowerCase())
+                        );
+                    });
+
+                    filteredItem.children = filteredChildren;
+                }
+
+                return filteredItem;
+            });
+
+            setFilteredItems(filtered);
+            setIsFiltered(true);
+        }
+    }
+
+    function onSelect(evt, treeViewItem) {
+        // Ignore folders for selection
+        if (treeViewItem && !treeViewItem.children) {
+            // setActiveItems([treeViewItem]);
+            const itemKey = treeViewItem.title;
+            const itemToAdd = descriptors.find(
+                (descriptor) => descriptor.shortName === itemKey || descriptor.name === itemKey
+            );
+            const newPolicyFieldCard = getEmptyPolicyFieldCard(itemToAdd);
+
+            addPolicyFieldCardHandler(newPolicyFieldCard);
+        }
+    }
+
+    function addPolicyFieldCardHandler(fieldCard) {
+        setFieldValue(`policySections[${selectedSectionIndex as unknown as string}].policyGroups`, [
+            ...policyGroups,
+            fieldCard,
+        ]);
+    }
+
+    const toolbar = (
+        <>
+            <Toolbar className="pf-v5-u-px-sm pf-v5-py-0">
+                <ToolbarContent style={{ padding: 0 }}>
+                    <ToolbarItem widths={{ default: '100%' }}>
+                        <TreeViewSearch
+                            onSearch={onSearch}
+                            id="input-search"
+                            name="search-input"
+                            aria-label="Filter policy criteria"
+                        />
+                    </ToolbarItem>
+                </ToolbarContent>
+            </Toolbar>
+            <Divider component="div" className="pf-u-mb-sm pf-u-mt-md" />
+        </>
+    );
+
+    return (
+        <Flex
+            className="pf-v5-u-w-100"
+            direction={{ default: 'column' }}
+            spaceItems={{ default: 'spaceItemsNone' }}
+        >
+            <Title headingLevel="h3" className="pf-v5-u-px-md pf-v5-u-pt-md">
+                Add policy criteria to{' '}
+                {selectedSection.sectionName || `Condition ${selectedSectionIndex + 1}`}
```

Comment:
I don't know if it is possible with the existing code paths, but `selectedSection` has a chance to be `undefined` here if indexing into the `values.policySections` array fails, which would lead to a crash during render. Worthwhile to be defensive here since we are likely never going to enable `noUncheckedIndexedAccess` in our TS config. 🫤 
```suggestion
                {selectedSection?.sectionName || `Condition ${selectedSectionIndex + 1}`}
```

Reply:
Aha - yeah if you load up the policy form and delete the only policy section it does lead to a page crash.

Actually this seems to be an issue whenever the _first_ section is deleted, regardless of the number of sections that are present.